server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregationBuilder.java

```diff
@@ -149,18 +122,18 @@ private void read(StreamInput in) throws IOException {
     @Override
     protected final void doWriteTo(StreamOutput out) throws IOException {
         if (serializeTargetValueType(out.getVersion())) {
-            out.writeOptionalWriteable(targetValueType);
+            out.writeOptionalWriteable(null);
```

Comment:
Based on my code analysis, we always wrote a null anyway, so this should be backwards compatible.  "Should" being the key word.

Reply:
Should we drop a TODO here to stop serializing it entirely in future versions?  

I wouldn't do that yet, there are enough balls in the air, but it'd be good to not forget about.

Reply:
I have a check box in the meta issue to come up with a deprecation plan for this, which is probably good enough for now.  But yes, I'd like to return to `serializeTargetValueType` before we merge to master.

Reply:
:+1: